lib/std/fmt.zig

```diff
@@ -1733,12 +1733,23 @@ pub fn Formatter(comptime format_fn: anytype) type {
 /// Ignores '_' character in `buf`.
 /// See also `parseUnsigned`.
 pub fn parseInt(comptime T: type, buf: []const u8, base: u8) ParseIntError!T {
-    if (buf.len == 0) return error.InvalidCharacter;
+    if (buf.len == 0) {
+        if (comptime isOptional(T)) return null;
```

Comment:
if think this could be simplified to 
```zig
if (@typeInfo(T) == .Optional) return null;
```

Reply:
That's cool, I am new to zig, thanks for the advice.